src/app/page.tsx

```diff
@@ -1,19 +1,104 @@
 "use client";
 
 import React, { useState } from "react";
+import { FaRegCircleDot } from "react-icons/fa6";
+import { useSelector } from "react-redux";
 
 import BytecodeInput from "../components/BytecodeInput";
+import DeployButton from "../components/DeployButton";
+import DeployModal from "../components/DeployModal";
+import { chainList } from "../data/chains";
+import { SupportedChain } from "../interfaces/data/chains";
+import { RootState } from "../state/store";
 
 export default function Home() {
+	const { isWalletConnected, currentNetwork, nativeBalance } = useSelector(
+		(state: RootState) => state.wallet
+	);
 	const [bytecode, setBytecode] = useState<string>("");
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+	const [selectedChain, setSelectedChain] = useState<SupportedChain | null>(
+		null
+	);
+	const [homeErrorMessage, setHomeErrorMessage] = useState<string>("");
+
+	const closeModal = () => {
+		setSelectedChain(null);
+		setIsModalOpen(false);
+	};
+
+	const openModal = (chain: SupportedChain) => {
+		setSelectedChain(chain);
+		setIsModalOpen(true);
+	};
+
+	/**
+	 * TODO: Need to somehow know if contract has been deployed to chain.
```

Comment:
The contract should return the child contract's address. For verification, upon receiving the address, you code reach out to the chain for that address and then check that the account has nonzero `bytecode` associated with it.

Moreover, the `xsafe-contracts` PR under review adds deployed contracts to a mapping, which the frontend will be able to query from each chain.

Aside from that, remove this `TODO` comment from the code.